src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/AgencyIntegrationTest.kt

```diff
@@ -1,54 +1,128 @@
 package uk.gov.justice.digital.hmpps.whereabouts.integration
 
-import org.assertj.core.api.Assertions.assertThat
 import org.junit.jupiter.api.Test
-import org.springframework.boot.test.web.client.exchange
-import org.springframework.http.HttpMethod
-import org.springframework.http.ResponseEntity
-import uk.gov.justice.digital.hmpps.whereabouts.dto.ErrorResponse
 
 class AgencyIntegrationTest: IntegrationTest() {
 
   @Test
-  fun `location groups - none in properties - should retrieve groups from elite2api`() {
+  fun `location groups - none in properties - should retrieve groups from prison API`() {
     val agencyId = "LEI"
     prisonApiMockServer.stubGetAgencyLocationGroups(agencyId)
 
-    val response: ResponseEntity<String> =
-        restTemplate.exchange("/agencies/$agencyId/locations/groups", HttpMethod.GET, createHeaderEntity(""))
+    webTestClient.get()
+        .uri("/agencies/$agencyId/locations/groups")
+        .headers(setHeaders())
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$.[0].name").isEqualTo("Block A")
+        .jsonPath("$.[0].key").isEqualTo("A")
+        .jsonPath("$.[0].children").isEmpty
 
-    assertThatJsonFileAndStatus(response, 200, "LEI_location_groups.json");
   }
 
   @Test
   fun `location groups - exist in properties - should retrieve groups from properties`() {
     val agencyId = "MDI"
 
-    val response: ResponseEntity<String> =
-        restTemplate.exchange("/agencies/$agencyId/locations/groups", HttpMethod.GET, createHeaderEntity(""))
-
-    assertThatJsonFileAndStatus(response, 200, "MDI_location_groups.json");
+    webTestClient.get()
```

Comment:
I think it looks much neater with those json files. What do you think? Might be worth tweaking assertThatJsonFileAndStatus later on at some point if you agree.

Reply:
I generally agree, and I tried to get it budge, but didn't succeed. I think this particular test is an extreme example, but it is likely we might have other long ones. Haven't deleted the JSON files for that very reason.